packages/hint/src/lib/types/new-async-html.ts

```diff
@@ -0,0 +1,97 @@
+import * as parse5 from 'parse5';
+import * as htmlparser2Adapter from 'parse5-htmlparser2-tree-adapter';
+import * as cssSelect from 'css-select';
+
+import {
+    IAsyncHTMLElement,
+    IAsyncHTMLDocument,
+    IAsyncNamedNodeMap,
+    ProblemLocation
+} from '../types';
+
+export class NewAsyncHTMLElement implements IAsyncHTMLElement {
+    public ownerDocument: IAsyncHTMLDocument;
+
+    private _element: Element;
+
+    public constructor(element: Element, ownerDocument: IAsyncHTMLDocument) {
+        this._element = element;
+        this.ownerDocument = ownerDocument;
+    }
+
+    public get attributes(): IAsyncNamedNodeMap {
+        const x = (this._element as any).attribs;
+
+        return Object.entries(x).map((y) => {
+            return {
+                name: y[0],
+                value: y[1] as string
+            };
+        }) as IAsyncNamedNodeMap;
+    }
+
+    public get nodeName(): string {
+        return (this._element as any).tagName;
+    }
+
+    public getAttribute(attribute: string): string | null {
+        const value = (this._element as any).attribs[attribute] || null;
+
+        return value;
+    }
+
+    public getLocation(): ProblemLocation | null {
+        const location = (this._element as any).sourceCodeLocation;
+
+        if (!location) {
+            return null;
+        }
+
+        return {
+            column: location.startCol,
+            line: location.startLine
+        };
+    }
+
+    public isSame(element: NewAsyncHTMLElement): boolean {
+        return this._element === element._element;
+    }
+
+    public outerHTML(): Promise<string> {
+        return Promise.resolve(parse5.serialize(this._element, { treeAdapter: htmlparser2Adapter }));
```

Comment:
Going back to the subject of making it `sync` again, this should be removed (same in the rest of the file).

Reply:
what do you mean for 'remove'? do you mean the Promise.resolve thing or the whole outerHTML?

Reply:
Remove the `Promise.resolve` so it is sync. Sorry about that.

Reply:
Ah ok, yes, there is some `Promise.resolve` to remove :)